src/det.jl

```diff
@@ -21,6 +21,23 @@ end
     return vecdot(x0, cross(x1, x2))
 end
 
+@inline function _det(::Size{(4,4)}, A::StaticMatrix,S::Type)
+    A = similar_type(A,S)(A)
```

Comment:
Or move the promoting to the dispatching function all together and dropping the propagation of the promoted type.
Better in my opinion

Both will work , what style do you prefer?